io/disk/Avago_storage_adapter/avago9361_vd.py

```diff
@@ -84,6 +84,20 @@ def setUp(self):
         self.stripe = [64, 128, 256, 512, 1024]
         self.state = ['start', 'stop', 'start', 'pause', 'resume']
         self.list_test = ['online_offline', 'rebuild']
+        cmd = "%s /c%d set jbod=on" % (self.tool, self.controller)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.log.info(" Good.. JBOD mode already set, Ignore the Error as it is pre-setup")
+        time.sleep(5)
+        cmd = "%s /c%d/vall delete force" % (self.tool, self.controller)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.log.info(" As required,  No Virtual Drives are Available. Ignore the Error as it is pre-setup")
+        cmd = "%s /c%d set jbod=off" % (self.tool, self.controller)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.log.info("As required, Controller is already in non-JBOD mode!!, Ignore the Error as it is pre-setup")
+        cmd = "%s /c%d/eall/sall set good force" % (self.tool, self.controller)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.log.info("As required Drives are Already Set to good!!, Ignore the Error as it is pre-setup")
+        time.sleep(10)
```

Comment:
Can we have separate functions for jbod enable and disable, so that we can re-use in `test_jbod` also ?

Reply:
Also, I would suggest going for better options than sleep(X)

Reply:
@manvanthar you can use avocado `wait.py` wait module to wait for command to complete. like

`wait.wait_for(self.check_pass(cmd, errmsg), 10)`

Reply:
@abdhaleegit   The problem is not with command completion.   command completes immediatly.

  The card takes some random sec to finish of its work before accepting the next of commands.  If we dont wait for few seconds,  the next set of commands fails intermitently.

Reply:
as discussed, the wait here is required for the internal state change of the device, and user could not track the internal state change.. so wait here is valid.